articles/custom_storage_location.md

```diff
@@ -144,157 +132,247 @@ For read-write permissions (you allow Synapse to upload and retrieve files)
+### Set S3 Bucket as Upload Location
 
-<img src="/assets/images/policy-dialog-paste.jpg">
+By default, your `Project/Folder` uses Synapse storage. You can use the external bucket configured above via our programmatic clients or web client.
 
-____
+{% tabs %}
 
+{% tab Python %}
+{% highlight python %}
+# Set storage location
+import synapseclient
+import json
+syn = synapseclient.login()
+PROJECT = 'syn12345'
+
+destination = {'uploadType':'S3', 
+               'concreteType':'org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
+               'bucket':'nameofyourbucket'}
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
 
-#### 5. Create an object that proves to the Synapse service that you own this bucket
-Open a new file in notepad and type in your synapse user name
+project_destination ={'concreteType': 'org.sagebionetworks.repo.model.project.UploadDestinationListSetting', 
+                      'settingsType': 'upload'}
+project_destination['locations'] = [destination['storageLocationId']]
+project_destination['projectId'] = PROJECT
+
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
+{% endhighlight %}
+{% endtab %}
 
-<img src="/assets/images/notepad.jpg">
+{% tab R %}
+{% highlight r %}
+#set storage location
+library(synapseClient)
+synapseLogin()
+projectId <- 'syn12345'
 
-and save the file as **owner.txt**
+destination <- list(uploadType='S3', 
+                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
+                    bucket='nameofyourbucket')
+destination <- synRestPOST('/storageLocation', body=destination)
 
-<img src="/assets/images/save-as-owner.jpg">
+projectDestination <- list(concreteType='org.sagebionetworks.repo.model.project.UploadDestinationListSetting', 
+                           settingsType='upload')
+projectDestination$locations <- list(destination$storageLocationId)
+projectDestination$projectId <- projectId
 
-click on the new bucket name to open up your new bucket
+projectDestination <- synRestPOST('/projectSettings', body = projectDestination)
+{% endhighlight %}
+{% endtab %}
 
-<img src="/assets/images/select-bucket.jpg">
+{% tab Web %}
+ Navigate to your **Project/Folder -> Tools -> Change Storage Location**. In the resulting pop-up, select the `Amazon S3 Bucket` option and fill in the relevant information, where Bucket is the name of your external bucket, Base Key is the name of the folder in your bucket to upload to, and Banner is a short description such as who owns the storage location:
 
-and click the **Upload** button
+<img id="image" src="/assets/images/external_s3.png">
 
-<img src="/assets/images/click-upload.jpg">
+{% endtab %}
 
-to show the upload dialog
+{% endtabs %}
 
-<img src="/assets/images/upload-dialog.jpg">
 
-click on **Add Files**
+<br/>
 
-<img src="/assets/images/click-add-file.jpg">
+### Example of uploading a file to S3 bucket
 
```

Comment:
If you are not permitting Synapse to write to your bucket or if you already have files that you would like represented in Synapse you will have to add them before they appear in Synapse.

This is done by cre...